client/src/components/TaskDurationBarChart/TaskDurationBarChart.js

```diff
@@ -0,0 +1,62 @@
+import React from "react";
+import styles from "./TaskDurationBarChart.module.css";
+import PropTypes from "prop-types";
+import { BarChart, Bar, XAxis, YAxis, ReferenceLine } from "recharts";
+
+const computeDurationDifference = (taskResponseData) => {
+  const taskDurationDifferences = [];
+
+  for (const task of taskResponseData) {
+    const taskDifference = {
+      taskName: task.title,
+      percentDifference: 0,
+    };
+
+    const plannedDuration =
+      task.originalEndDate.getTime() - task.originalStartDate.getTime();
+    const actualDuration = task.endDate.getTime() - task.startDate.getTime();
+    taskDifference.percentDifference =
+      (actualDuration - plannedDuration) / plannedDuration;
+
+    if (task.taskStatus === "DONE")
+      taskDurationDifferences.push(taskDifference);
+  }
+
+  return taskDurationDifferences;
+};
+
+export const TaskDurationBarChart = ({ taskResponseData }) => {
```

Comment:
For the `computeDurationDifference` function, I would still just keep this inside the TaskDuractionBarChart component since I don't think this function will be used elsewhere (or by any other components).

Generally I would keep my functions within my component

Reply:
Moved to inside the component.